### ISSUE TITLE
how to use SyncBN?

### ISSUE THREAD

**Issue by root12321** [open-mmlab/mmclassification#908]
how to use SyncBN when i training model in multi GPUS

**Comment by Ezra-Yu** [comment]
refer to https://github.com/open-mmlab/mmclassification/blob/d2e505415040bf5329ab218bb6fe3d899f176cd5/configs/resnet/resnet50_8xb256-rsb-a1-600e_in1k.py#L11

**Comment by root12321** [comment]
> 

thank you ,i can use SyncBN now.but i have a question ,when i train a model with single gpu and multi gpus, the result of multi gpus accuracy is less than single gpu about 5 scores, i think the differences is very big , how to solve this question?

**Comment by Ezra-Yu** [comment]
Do you use the same config? Can you provide your config and commands to test? 
And run ```pip list | grep "torch\|mmcv\|mmcls" to see the environment. I will check that.

Try to turn off the grad of SyncBN:
```
norm_cfg=dict(type='SyncBN', requires_grad=False), 
```

**Comment by Ezra-Yu** [comment]
I mean run 
```
CUDA_VISIBLE_DEVICES=7 python tools/est.py /home/leinao/zhangzhiyong/mmclassification/class_test/resnet-18.py xx.pth
```
and
```
CUDA_VISIBLE_DEVICES=6,7 ./tools/dist_test.sh /home/leinao/zhangzhiyong/mmclassification/class_test/resnet-18.py 2 xxx.pth
```
Is there difference?

**Comment by root12321** [comment]
> 

yes,the result is same
![image](https://user-images.githubusercontent.com/28287748/178221188-af1b9367-88d9-4618-a507-0f5a981d506b.png)
![image](https://user-images.githubusercontent.com/28287748/178221237-998b1072-c32b-4f2e-8d83-3e5a96ef24a3.png)
@Ezra-Yu

**Comment by Ezra-Yu** [comment]
Now, the problem is that the acc is lower when using `SyncBN`, since `SyncBN` has no effect when using only one GPU. 

This may be caused by your lr setting, try to scale your lr when trying on multi GPUs. In your case:
1 GPU
```
optimizer = dict(type='SGD', lr=0.002, momentum=0.9, weight_decay=0.0001)
```

N GPU
```
optimizer = dict(type='SGD', lr=0.002 * N, momentum=0.9, weight_decay=0.0001)
```

**Comment by root12321** [comment]
> Now, the problem is that the acc is lower when using `SyncBN`, since `SyncBN` has no effect when using only one GPU.
> 
> This may be caused by your lr setting, try to scale your lr when trying on multi GPUs. In your case: 1 GPU
> 
> ```
> optimizer = dict(type='SGD', lr=0.002, momentum=0.9, weight_decay=0.0001)
> ```
> 
> N GPU
> 
> ```
> optimizer = dict(type='SGD', lr=0.002 * N, momentum=0.9, weight_decay=0.0001)
> ```
yes,i tryed,when i use single gpu, i set lr=0.001,batch_size=32,when i use 2 GPUs, i set lr=0.002,batch_size=64, `norm_cfg=dict(type='SyncBN', requires_grad=False)` 
but  result is not changed

**Comment by Ezra-Yu** [comment]
How about using normal BatchNorm?

I think I can not solve your question, Maybe, Next, you need to design your own experiments to verify.

**Comment by root12321** [comment]
normal BatchNorm is also can not solve the question 

thank you ,may be i need to search other solutions
